src/realtime.py

```diff
@@ -22,90 +23,152 @@
+            return None
+
+        # Read and convert data
+        fields = base_filename.split(".")
+        file_time = fields[0] + fields[1] + fields[2] + fields[3]
+
+        # Make sure we only process the first slice for simultaneous multi-slice data for now
+        if file_time == self.last_file_time:
+            self.log.debug("not processing multi-slice data for slices > 0", filename=base_filename)
+            os.remove(filename)
+            return None
+
+        self.last_file_time = file_time
+
+        slice_num = int(fields[5])
+
+        try:
+            self.log.info("using pydarnio to convert", filename=filename)
+            converted = pydarnio.BorealisConvert(filename, "rawacf", "/dev/null", slice_num, "site")
+            os.remove(filename)
+        except pydarnio.exceptions.borealis_exceptions.BorealisConvert2RawacfError as e:
+            self.log.warn("error converting", filename=filename)
+            os.remove(filename)
+            return None
+
+        return fitacf._fit(converted.sdarn_dict[0])
+
+    def fitacf_data_to_queue(self, fitacf_data):
+        """
+        Makes a copy of the fitacf data, and converts to native python type.
+        Adds converted fitacf data to queue.
+        :param fitacf_data: fitacf data as output from backscatter
+        """
+        tmp = fitacf_data.copy()
+
+        # Can't jsonify numpy, so we convert to native types for rt purposes.
+        for k, v in fitacf_data.items():
+            if hasattr(v, 'dtype'):
+                if isinstance(v, np.ndarray):
+                    tmp[k] = v.tolist()
+                else:
+                    tmp[k] = v.item()
+        # q.put(item) by default blocks until a slot is available in the queue
+        self.q.put(tmp)
+
+    def get_temp_file_from_datawrite(self):
+        """
+        Runs infinite loop waiting for a file name received on socket from datawrite module.
+        If the file is a rawacf file, then it converts the rawacf to fitacf data, and adds
+        the data to the queue.
+        """
+        while True:
+            filename = so.recv_data(self.data_write_to_realtime, self.options.dw_to_rt_identity,
+                                    self.log)
+
+            fitacf_data = self.read_and_convert_file_to_fitacf(filename)
+            if fitacf_data:
+                self.fitacf_data_to_queue(fitacf_data)
+
+    def handle_remote_connection(self):
+        """
+        Compresses and serializes the data to send to the client.
+        """
+        while True:
+            # q.get() default blocks until an item is available
+            data_dict = self.q.get()
+            serialized = json.dumps(data_dict)
+            compressed = zlib.compress(serialized.encode())
+            self.realtime_socket.send(compressed)
 
 
 if __name__ == '__main__':
     from utils import log_config
     log = log_config.log()
     log.info(f"REALTIME BOOTED")
     try:
-        main()
+        rt = RealtimeServer(log)
```

Comment:
Within the same module you should not need to pass in the logger into your functions or classes. I think you told me you did this, so logs would work when running unittest? What should be done is untitest should have its own logger so that when you import this class to test, the testing logs will go to the testers log file. Was that not working?

Reply:
No I only did this because I didn't know that you didn't need to pass it in.
Wasn't thinking about unittest, and didn't try using the testing.